### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by ffrank**
Prometheus tests are busted.

Make their output more readable, and make sure they don't leave lingering mgmt processes upon failing.

### CODE REVIEW

**Comment by ffrank** [comment]
This is WiP.

Also please disregard the git thing, it got in here by accident and will be rebased away in time 🙈

**Comment by ffrank** [comment]
Ready for merge.

It fixes the consistent test failures. The etcd race is not yet addressed, but will be easier to debug, and actual regressions will be more visible again.

File: .gitmodules [hunk 1]
```diff
@@ -15,7 +15,7 @@
 	url = https://github.com/purpleidea/go-systemd
 [submodule "vendor/honnef.co/go/augeas"]
 	path = vendor/honnef.co/go/augeas
-	url = https://github.com/dominikh/go-augeas/
```
Comment (purpleidea) [inline .gitmodules]: Why are we removing the trailing slashes?

File: test/shell/exec-usergroup.sh [hunk 2]
```diff
@@ -76,3 +76,6 @@ cleanup
 run_usergroup_test "exec-usergroup-${GROUP}.yaml" "nobody" "${GROUP}"
 run_usergroup_test "exec-usergroup-user.yaml" "nobody" ""
 run_usergroup_test "exec-usergroup-group-${GROUP}.yaml" "" "${GROUP}"
+
+# avoid race against rm command from the shell test wrapper
```
Comment (purpleidea) [inline test/shell/exec-usergroup.sh]: What race is that?

File: test/shell/load0.sh [hunk 3]
```diff
@@ -33,7 +33,7 @@ import "sys"
 
 file "${tmpdir}/loadavg" {
 	content => fmt.printf("load average: %f, %f, %f\n", \$x1, \$x5, \$x15),
-	state => $const.res.file.state.exists,
+	state => \$const.res.file.state.exists,
```
Comment (purpleidea) [inline test/shell/load0.sh]: Nice catch, thanks!

File: test/shell/prometheus-2.sh [hunk 4]
```diff
@@ -2,17 +2,39 @@
 
 . "$(dirname "$0")/../util.sh"
 
+TEMPFILE=`mktemp mgmt-tests-XXXXXXXX`
+
+cleanup()
```
Comment (purpleidea) [inline test/shell/prometheus-2.sh]: I'm fine with having these here, but would putting them in the above listed util.sh help, and/or is there something there that's already useful?

File: .gitmodules (hunk 1 above)
Comment (ffrank) [reply .gitmodules]: Some git compat issue, but this is no longer part of this patchset anyway.

File: test/shell/exec-usergroup.sh (hunk 2 above)
Comment (ffrank) [reply test/shell/exec-usergroup.sh]: Not 100% certain it's a race, but I saw this spectacular failure: https://travis-ci.org/purpleidea/mgmt/jobs/647560853?utm_medium=notification&utm_source=github_status

**Comment by purpleidea** [comment]
Okay, thanks! I've merged 2/3 of the commits via cherry-pick to git master. The submodules one should be changed I guess, but not sure why the trailing slashes are being changed.

**Comment by ffrank** [comment]
I have no recollection of what was going on here, and cannot imagine it might be of value today. Closing.

**Pull request closed without merging**